packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -167,17 +203,32 @@ export class SelectedNetworkController extends BaseController<
 
     this.update((state) => {
       state.domains[domain] = networkClientId;
-      if (!state.perDomainNetwork) {
-        state.domains[METAMASK_DOMAIN] = networkClientId;
-      }
     });
   }
 
+  #domainHasPermissions(domain: Domain): boolean {
+    return this.messagingSystem.call(
+      'PermissionController:hasPermissions',
+      domain,
+    );
+  }
+
   setNetworkClientIdForDomain(
     domain: Domain,
     networkClientId: NetworkClientId,
   ) {
-    if (!this.state.perDomainNetwork) {
+    // Early return if perDomainNetwork is disabled and the domain is not Metamask, unless it's setting for Metamask specifically
```

Comment:
i believe the if statement is clear enough already

Reply:
removed here: https://github.com/MetaMask/core/commit/ecb5e72a77608459747c279a2e14aad4c94b63a1